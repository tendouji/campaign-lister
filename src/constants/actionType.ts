export const STORE_CAMPAIGN_DATA = 'STORE_CAMPAIGN_DATA';
export const SET_SEARCH_KEY = 'SET_SEARCH_KEY';
export const SET_DATE_RANGE = 'SET_DATE_RANGE';
export const SET_SNACKBAR_DISPLAY = 'SET_SNACKBAR_DISPLAY';
export const SET_MODALPOPUP_DISPLAY = 'SET_MODALPOPUP_DISPLAY';
