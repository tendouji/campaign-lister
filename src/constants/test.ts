import React, {ReactElement} from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";


const testLabels = {
    ComponentMatchSnapshot: (componentName: string) => `render to match snapshot for ${componentName} component`,
    ComponentRenderWithoutCrash: (componentName: string) => `render without crashing for ${componentName} component`,
};

const testFunctions = {
    ComponentMatchSnapshot: (component: ReactElement): any => {
        const cloneComponent = React.cloneElement(component);
        const Component = renderer.create(cloneComponent).toJSON();
        expect(Component).toMatchSnapshot();
    },
    ComponentRenderWithoutCrash: (component: ReactElement): any => {
        const div = document.createElement('div');
        ReactDOM.render(component, div);
        ReactDOM.unmountComponentAtNode(div);
    }
};


export {
    testLabels,
    testFunctions,
}