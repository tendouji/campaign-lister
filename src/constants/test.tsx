import React, {ReactElement} from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { appReducer } from "../reducers";
import App from "../App";
import {render, shallow} from "enzyme";

const store = createStore(appReducer);

const testLabels = {
    ComponentMatchSnapshot: (componentName: string) => `render to match snapshot for ${componentName} component`,
    ComponentRenderWithoutCrash: (componentName: string) => `render without crashing for ${componentName} component`,
};

const testFunctions = {
    ComponentMatchSnapshot: (component: ReactElement, withStore: boolean | undefined = undefined): any => {
        const cloneComponent = React.cloneElement(component);

        let _cloneComponent: ReactElement = (!!withStore
            ? <Provider store={store}>{cloneComponent}</Provider>
            : cloneComponent);

        const Component = renderer.create(_cloneComponent).toJSON();
        expect(Component).toMatchSnapshot();
    },
    ComponentRenderWithoutCrash: (Component: ReactElement, withStore: boolean | undefined = undefined): any => {
        const wrapper = render(Component);

        expect(wrapper.html()).toContain('div');


        const div = document.createElement('div');
        ReactDOM.render(<Provider store={store}>
            { Component }
        </Provider>

        , div);
        ReactDOM.unmountComponentAtNode(div);
    }
};


export {
    testLabels,
    testFunctions,
}