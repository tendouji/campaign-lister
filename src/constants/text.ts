const footerString = (year: number): string => `Â© ${year} Chan Wan / CR8 Campaign`;

export {
    footerString,
}