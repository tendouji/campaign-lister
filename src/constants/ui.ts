const colors = {
    primary: '#43cadb',
    white: '#fff',
    black: '#000',
    darkGray: '#666',
    gray: '#ddd',
    lightGray: '#f6f6f6',
    lightGray2: '#eee',
    green: '#4caf50',
    red: '#ff2717',
    yellow: '#ffc107',
    blue: '#03a9f4',
};

const layout = {
    xSmallGap: '0.3125rem',
    smallGap: '0.625rem',
    standardGap: '1rem',
    largeGap: '1.25rem',
    borderRadius: '4px',

    fontSizeSmall: '0.75rem',
    fontSizeStandard: '0.875rem',
    fontSizeMedium: '1rem',
    fontSizeLarge: '1.25rem',
    fontSizeXLarge: '1.5rem',

    filterBarHeight: '4.5rem',
};

const styles = {
    lineStyle: `1px solid ${colors.gray}`,
    textOverflowStyle: `
        white-space: nowrap;
        text-overflow: ellipsis;
        overflow: hidden;
    `,

    filterBarButtonStyle: `
        width: ${layout.filterBarHeight};
        height: ${layout.filterBarHeight};
        padding: 0;
        border: 0;
        font-size: ${layout.fontSizeXLarge};
        cursor: pointer;
        
        &:hover {
            background-color: ${colors.lightGray};
        }
        
        &:focus {
            outline: none;
        }
    `
};


export {
    colors,
    layout,
    styles,
}