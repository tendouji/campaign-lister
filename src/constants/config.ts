import {campaignDataLabels} from "./text";

export const config = {
    minSearchLength: 3,
    recordPerPage: 5,

    snackBarLifeSpan: 5000,
    snackBarTransition: 250,

    mediumScreenWidth: '37.5rem',
    smallScreenWidth: '25rem',

    campaignDataLabels: {
        name: campaignDataLabels.name,
        startDate: campaignDataLabels.startDate,
        endDate: campaignDataLabels.endDate,
        budget: campaignDataLabels.budget,
        status: campaignDataLabels.status,
    }
};