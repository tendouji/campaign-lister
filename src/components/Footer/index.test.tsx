import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { footerString } from "../../constants/text";
import { Footer } from './';


describe('<Footer />', () => {
    const wrapper = shallow(<Footer />);

    it('renders without crashing', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('renders without crashing', () => {
        const currentYear: number = (new Date()).getFullYear();
        expect(wrapper.text()).toEqual(footerString(currentYear));
    });
});


