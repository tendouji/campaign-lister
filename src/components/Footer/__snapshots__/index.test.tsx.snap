// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders without crashing 1`] = `
<Fragment>
  <styled.div />
  <styled.footer
    className="footer"
  >
    <span>
      Â© 2019 Chan Wan / CR8 Campaign
    </span>
  </styled.footer>
</Fragment>
`;
